File: 3740.minimum-distance-between-three-equal-elements-i/solution.kt
class Solution {
    fun minimumDistance(nums: IntArray): Int {
        var res = Int.MAX_VALUE
        val first = mutableMapOf<Int, Int>()
        val second = mutableMapOf<Int, Int>()

        nums.forEachIndexed { i, n ->
            if (n in second) {
                res = minOf(res, 2 * (i - first.getValue(n)))
                first[n] = second.getValue(n)
                second[n] = i
            } else if (n in first) {
                second[n] = i
            } else {
                first[n] = i
            }
        }

        return if (res == Int.MAX_VALUE) -1 else res
    }
}

File: 3740.minimum-distance-between-three-equal-elements-i/solution.go
func minimumDistance(nums []int) int {
    res := math.MaxInt
    for i := range nums {
        for j := i+1; j < len(nums); j++ {
            if nums[j] == nums[i] {
                for k := j+1; k < len(nums); k++ {
                    if nums[k] == nums[i] {
                        res = min(res, 2*(k-i))
                    }
                }
            }
        }
    }
    if res == math.MaxInt {
        return -1
    }
    return res
}